### ISSUE TITLE
Check for door ajar before accepting resume or jog commands

### ISSUE THREAD

**Issue by sharmstr** [OpenBuilds/OpenBuilds-CONTROL#92]
Hey Peter.  I've been playing around with the door ajar stuff.  Is there any way you can not allow resume if the door pin is in a triggered state?  How about disabling jog buttons as well?  Here's what's happening in my test.

1 - Issue a movement command (pressing Y+ 100mm)
2 - During movement press my door button
3 - Before releasing the button, I can press the resume button. The gantry doesnt move (becuase door pin is still triggered), but Control recognizes the press so it grays out resume button.
4 - Release door pin, nothing happens and cant click resume again because its grayed out.
5 - I can click the jog buttons.  Nothing happens (which is good), but it saves those commands (which is bad, I think)
6 - Trigger the door pin again which makes resume button available again.
7 - Release door pin before clicking resume.  
8 - Click resume and now all those jog commands I pressed before are processed.

Basically I'm looking for better handling of a door pin triggered condition.  Maybe disable and  turn the resume button Red while door pin is triggered.  Flash the resume button after door pin is cleared.  All the while disabling all types of movement via UI and keyboard (jog, home, goto, whatever else I'm missing)

**Comment by petervanderwalt** [comment]
https://openbuilds.com/threads/resume-job-automatically.15818/#post-96517 refers

From How does the "Safety Door" features work? · Issue  grbl/grbl#713

> The safety door will disable the spindle and coolant outputs and will only allow you to resume via '~' once the switch is closed. > It'll repower the spindle and coolant after a configurable dwell before resuming the program.

So actually we don't need to look at D to pause or not.We pause "if (pins.includes('H'))" regardless

The difference lies in how to resume... Play button used to be available whether door was open or closed after feedhold. But it should have only been an option once the door is closed again! That's what caused the issues. So in upcoming version 1.0.238 or later

- Open the door, you get H, which Pauses the UI (pause button switches to a play button, statusses show paused etc)
- We also get a D, so we Disable the RESUME button
- Only once the D goes away from pin Status, do we enable Resume, allowing you to, as per above "only allow you to resume via '~' once the switch is closed"
Otherwise the resume used to be ignored and then state thinks its playing, when it is still paused

Be sure to test 1.0.238 when it arrives and let us know if that resolved it
